source/manuals/programming-languages/html.html.md.erb

```diff
@@ -1,85 +1,153 @@
-  accessibility
+The GOV.UK Service Manual lists the [minimum browsers that GDS is required
+to support](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in).
+Tools such as GOV.UK Frontend, that are used to help build services, may have
+[stricter standards of browser support](https://github.com/alphagov/govuk-frontend#browser-support).
+Therefore, you will need to test your code in different browsers depending on
+which part of GDS you are working on.
 
-## Use [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) roles appropriately
+Whatever your level of browser support, HTML should be written in such a way
+that older browsers see an appropriate fallback or warning message. For example,
+some browsers do not support the `<video>` element, so you should ensure that
+some contextual information is included (the `<p>` element in this example):
 
-### Guidance on using [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmarks
+```html
+<video width="320" height="240" controls>
+  <source src="movie.mp4" type="video/mp4">
+  <source src="movie.ogg" type="video/ogg">
+  <p>Unable to load video. Please upgrade your browser.</p>
+</video>
+```
 
-When you use HTML5 sectioning elements it's a good idea to include
-the appropriate [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmark roles as well. The trick when you're
-using both HTML5 and [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) is to put the landmark on the sectioning
-element itself because this avoids the information being duplicated by
-screen readers that support both technologies.
+When using structual HTML5 elements, such as `<main>`, ensure that you include
+a [shim that allows them to be styled](https://github.com/aFarkas/html5shiv) in
+legacy browsers. You can include the script inside a `<!--[if lt IE 9]><![endif]-->`
+special comment so that modern browsers do not download them.
 
-Example:
+Your page should work with just HTML, before adding any CSS or JavaScript. Read
+[building a resilient frontend using progressive enhancement](https://www.gov.uk/service-manual/technology/using-progressive-enhancement).
+
+## Document structure
 
-    <header role="banner"></header>
+Use HTML5 to structure your page semantically. Use ARIA roles to help
+older assistive technologies map the sections correctly.
 
-The following roles map directly to HTML5 sectioning elements:
+Léonie Watson explains [why using ARIA landmark roles in this way delivers the
+best experience for screen reader users](http://tink.uk/screen-readers-aria-roles-html5-support/).
 
-* `role="main"` maps to the `<main>` element
-* `role="complementary"` maps to the `<aside>` element
-* `role="contentinfo"` maps to the `<footer>` element
-* `role="navigation"` maps to the `<nav>` element
+This document doesn't prescribe whether to use single or double quotes in attributes,
```

Comment:
I think that lines 47 to 50 are a bit too fussy and could be replaced with 'The project should be consistent in its code style'.

Reply:
I like that the current version is unambiguous - when I first joined, I wondered what the house style was in terms of quotes and self-closing elements and so on. I think "should be consistent in its code style" doesn't necessarily reassure me that, say, it's ok to not apply trailing slashes.

Reply:
I wasn't in the frontend meeting mentioned so not sure about what was decided but don't the deleted points that started on line 11 define a style? Excuse if their deletion was a decision already made.

Also worth noting that we use a fair amount of testing tools which parse our HTML using XML parsing libraries so not quoting attributes or closing elements (making them valid XML) would limit our use of them.

Reply:
@tombye I followed the original guide when writing some HTML and was wondering whether to use single or double quotes, and there were other files in the project that used no quotes at all (so apparently not following the guidance).

In trying to get an answer, it was discussed about 2 months ago at the frontend meeting and the collective decision was that consistency is the most important thing and that we shouldn't prescribe the style.

You make a good point about tooling. I suppose if we're not prescribing the style then the developer is free to adopt whatever style that is demanded by their tooling.